docs/Configuration/Channel-Drivers/WebSocket.md

```diff
@@ -0,0 +1,65 @@
+# WebSocket **DRAFT**
+
+## Background
+
+The WebSocket Channel Driver (chan_websocket) is designed to ease the burden on ARI application developers with getting media in and out of Asterisk.  The ARI /channels/externalMedia REST endpoint already has two other channel drivers  available (AudioSocket and RTP) but they require binary packet manipulation (RTP especially) and both require that the app developer handle the timing of sending packets to asterisk.  chan_websocket requires neither.
+
+## Features
+
+* Send and receive media using most Asterisk codecs.
+* Send arbitrary packet lengths to Asterisk.  The channel driver will break them up into appropriately sized frames (see notes below though).
+* No need to time your own packet transmits.
+* Silence is automatically generated when no packets have been received from the app.
+* The channel driver can accept incoming websocket connections _from_ your app as well as make outgoing connections _to_ your app.
+* Although the driver is targetted at ARI ExternalMedia users, it's not tied to ARI and can be used directly from the Dial dialplan app.
+
+## Details
+
+### Outgoing Connections
+
+Outgoing connections require you to pre-configure a websocket client in the `websocket_client.conf` config file (see details below).  Once done, you can reference the connection in a dial string.
+
+```ini
+[default]
+exten = _x.,1,Dial(WebSocket/connection1/c(ulaw))
+```
+
+This would connect to your application's websocket server using the client named `connection1` and using the `ulaw` codec.  When your server accepts the connection, the websocket channel will be answered and media will begin to flow.
+
+### Incoming Connections
+
+Incoming connections must be made to the global Asterisk HTTP server using the `media` URI but you must still 'Dial" the channel using the special `incoming` connection name.
+
+```ini
+[default]
+exten = _x.,1,Dial(WebSocket/incoming/c(ulaw))
+```
+
+The websocket channel will be created immediately and the `MEDIA_WEBSOCKET_CONNECTION_ID` channel variable will be set to an ephemeral connection id which must be used in the URI your application will connect to Asterisk with.  For example `/media/32966726-4388-456b-a333-fdf5dbecc60d`.  When Asterisk accepts the connection, the websocket channel will be answered and media will flow.
```

Comment:
Clarify what happens if you connect a second time.

Reply:
Done